src/control/server/server_utils.go

```diff
@@ -104,19 +104,63 @@ func writeCoreDumpFilter(log logging.Logger, path string, filter uint8) error {
 	return err
 }
 
-func createListener(ctlPort int, resolver resolveTCPFn, listener netListenFn) (*net.TCPAddr, net.Listener, error) {
-	ctlAddr, err := resolver("tcp", fmt.Sprintf("0.0.0.0:%d", ctlPort))
+func getNetInterfaceAddrs(ifaceName string) ([]net.Addr, error) {
+	iface, err := net.InterfaceByName(ifaceName)
 	if err != nil {
-		return nil, nil, errors.Wrap(err, "unable to resolve daos_server control address")
+		return nil, err
+	}
+
+	return iface.Addrs()
+}
+
+type ctlAddrParams struct {
+	iface         string
+	port          int
+	getIfaceAddrs func(string) ([]net.Addr, error)
+	resolveAddr   resolveTCPFn
+}
+
+func getControlAddr(params ctlAddrParams) (*net.TCPAddr, error) {
+	ipStr := "0.0.0.0"
+	if params.iface != "" {
+		addrs, err := params.getIfaceAddrs(params.iface)
+		if err != nil {
+			return nil, errors.Wrap(err, "getting interface addresses")
+		}
+
+		for _, addr := range addrs {
+			switch t := addr.(type) {
+			case *net.IPNet:
+				// FIXME: At the moment the control plane makes some assumptions
+				// about the address being IPv4.
+				if ip4 := t.IP.To4(); ip4 != nil {
+					ipStr = ip4.String()
+					break
```

Comment:
this will only break out of the switch block, need to set flag to skip out of loop

Reply:
presuming you want to break out after the first match, not iterate until the last

Reply:
Oops, will fix.